app/connectors/propertyLinking/PropertyLinkConnector.scala

```diff
@@ -95,15 +95,12 @@ class PropertyLinkConnector @Inject()(config: ServicesConfig, http: WSHttp)(impl
           "requestTotalRowCount" -> pagination.requestTotalRowCount.toString)
     )
 
-    def validAgent(agent: OwnerAuthAgent): Boolean =
-      agent.status.fold(false) { status =>
-        representationStatusFilter.map(_.name.toUpperCase).contains(status.toUpperCase)
-      }
-
     // filter agents on representationStatus
-    ownerAuthResult.map(oar =>
-      oar.copy(authorisations = oar.authorisations.map(auth =>
-        auth.copy(agents = auth.agents.map(ags => ags.filter(ag => validAgent(ag)))))))
+    filterAgents(ownerAuthResult, representationStatusFilter)
+  }
+
+  def validAgent(agent: OwnerAuthAgent, representationStatusFilter: Seq[RepresentationStatus]): Boolean = {
+    representationStatusFilter.map(_.name.toUpperCase).contains(agent.status.toUpperCase)
```

Comment:
there is an easier way to write this. 
with ignore case.

Reply:
done